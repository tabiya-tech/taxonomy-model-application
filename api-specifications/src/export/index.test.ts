describe("Test the export module", () => {
  test("The export module can be required via the index", () => {
    // GIVEN the module
    // WHEN the module is required via the index
    expect(() => {
      // THEN Check if the module can be required without error
      expect(() => {
        require("./");
      }).not.toThrowError();
      // AND check if the various exports are defined
      expect(require("./").default.Schemas.POST.Request.Payload).toBeDefined();
      expect(require("./").default.Constants.MAX_PAYLOAD_LENGTH).toBeDefined();
      expect(require("./").default.Enums.POST).toBeDefined();
    }).not.toThrowError();
  });

  test("The export module matches the snapshot", () => {
    // GIVEN the module
    // WHEN the module is required via the index
    expect(() => {
      // THEN Check if the module can be required without error
      expect(require("./").default).toMatchSnapshot();
    }).not.toThrowError();
  });
});

export {};
