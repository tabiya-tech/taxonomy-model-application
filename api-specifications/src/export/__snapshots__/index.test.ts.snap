// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the export module The export module matches the snapshot 1`] = `
{
  "Constants": {
    "MAX_PAYLOAD_LENGTH": 60,
  },
  "Enums": {
    "POST": {
      "Response": {
        "ExportResponseErrorCodes": {
          "FAILED_TO_TRIGGER_EXPORT": "FAILED_TO_TRIGGER_EXPORT",
        },
      },
    },
  },
  "Schemas": {
    "POST": {
      "Request": {
        "Payload": {
          "$id": "/components/schemas/ExportRequestSchemaPOST",
          "additionalProperties": false,
          "properties": {
            "modelId": {
              "description": "The identifier of the model to be exported.",
              "pattern": "^[0-9a-f]{24}$",
              "type": "string",
            },
          },
          "required": [
            "modelId",
          ],
          "type": "object",
        },
      },
    },
  },
  "Types": {},
}
`;
